src/main/java/org/skife/jdbi/v2/tweak/transactions/LocalTransactionHandler.java

```diff
@@ -228,6 +213,24 @@ public void setRollbackOnly()
         }
     }
 
+    private void restoreAutoCommitState(final Handle handle) {
+        try {
+            final LocalStuff stuff = localStuff.remove(handle);
+            if (stuff != null) {
+                handle.getConnection().setAutoCommit(stuff.getInitialAutocommit());
+                stuff.getCheckpoints().clear();
+            }
+        } catch (SQLException e) {
+            throw new UnableToRestoreAutoCommitStateException(e);
+        } finally {
+            // prevent memory leak if rollback throws an exception
+            if (localStuff.containsKey(handle)) {
```

Comment:
Checking `containsKey` before calling `remove` is redundant and should be removed